### PR TITLE
Addes new providers & corrected some

### PR DESCRIPTION
**Pull request by vfoucault**
This Pull request compile the previously created ones (and closed).

Fixed the `subserver` provider which wasn't working. It has some defects loading the `inetd.conf` resources and commands to create/update were wrong.
Corrected some issues also in the resource.

Added the following providers :

* `chdev` : udpate AIX devices attributes
* `chfs`: change AIX filesystem size (+/-)
* `user` : work with AIX users
* `group` : work with AIX groups

updated the `README.md` to include some details about the providers.

(Sorry, i had some troubles with the previous pull requests).

### CODE REVIEW

**Comment by juliandunn** [comment]
Hey @vfoucault this is pretty cool. Question about the user and group providers: wondering why we need those? Shouldn't the out-of-the-box Chef providers work?

File: README.md [hunk 1]
```diff
@@ -13,6 +13,140 @@ This cookbook has no recipes.
...
+```ruby
+aix_chfs "/tmp" do
+  action :resize
+  size 4
+  unit "G"
+end
+```
+valid actions : 
+
+* :rezie
```
Comment (juliandunn) [inline README.md]: :resize

File: providers/chdev.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#
+# Author:: Vianney Foucault (<vianney.foucault@gmail.com>)
+# Cookbook Name:: idp-aix
```
Comment (juliandunn) [inline providers/chdev.rb]: "aix"

File: providers/chdev.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# Provider:: aixchdev
+#
+# Copyright:: 2014, The Author
```
Comment (juliandunn) [inline providers/chdev.rb]: put your name or company name here.

File: providers/chdev.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        converge_by("Updating device #{@current_resource.name}") do
+          cmd = Mixlib::ShellOut.new(cmd)
+          cmd.valid_exit_codes = 0
+          cmd.run_command
+          cmd.error!
+          cmd.error?
+        end
+      end
+    end
+  end
```
Comment (juliandunn) [inline providers/chdev.rb]: There's some messed up indentation here.

**Comment by juliandunn** [comment]
Overall I like the work, though the patch is too big to take in one go. Can we break it up into two PRs?

* one to fix the subserver issues
* one to add filesystem management LWRPs

I'd also like to know what is wrong with the `user` and `group` providers in Chef core that necessitated you adding special providers here. We really should fix https://github.com/chef/chef/blob/master/lib/chef/provider/user/aix.rb and https://github.com/chef/chef/blob/master/lib/chef/provider/group/aix.rb respectively if there are missing capabilities.

**Comment by vfoucault** [comment]
Hi Julian,


First thanks for the reply.
Since those are overall my firsts recipe, i'm glad you like them.

I'll handle the re-reading and correcting of typing issues, and so push
them the way you recommend, subserver & chdev lwrp to AIX,
regarding the user & group providers, in AIX you do have a lot of option
that were not basically included the Chef::Provider::User::Aix, same thing
for group. So when i worked on my side, i thought that it could be easier
to write my owns. I guess that i will have to fully read the both user
provider & the aix one to make a patch, not that easy.

cheers.

**Comment by vfoucault** [comment]
Changes will be applied in the next PR.

**Pull request closed without merging**